### PR TITLE
feature: 33785 - Added the ability to retrieve inwoner count

### PR DESCRIPTION
**Pull request by Rick-Ritense**
Created a new GraphQL query for retrieving the inwoner count
Added the ability to actually retrieve them from Haal Centraal

Renamed and moved some classes to better differentiate between retrieving persons and bewoners

### CODE REVIEW

File: haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/graphql/HaalCentraalBrpQuery.kt [hunk 1]
```diff
@@ -28,4 +28,9 @@ class HaalCentraalBrpQuery(val haalCentraalBrpService: HaalCentraalBrpService) :
     suspend fun getPersoon(dfe: DataFetchingEnvironment): Persoon? {
         return haalCentraalBrpService.getPersoon(dfe.graphQlContext.get(SecurityConstants.AUTHENTICATION_KEY))
     }
+
+    @GraphQLDescription("Gets the amount of persons living in the same house as the person that makes the requests")
```
Comment (ivo-ritense) [inline haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/graphql/HaalCentraalBrpQuery.kt:32]: amount of persons -> number of people

File: haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/service/HaalCentraalBrpService.kt [hunk 2]
```diff
@@ -16,8 +16,11 @@
 package com.ritense.valtimo.portal.haalcentraal.brp.service
 
 import com.ritense.valtimo.portal.commonground.authentication.CommonGroundAuthentication
-import com.ritense.valtimo.portal.haalcentraal.brp.domain.Persoon
+import com.ritense.valtimo.portal.haalcentraal.brp.domain.bewoning.Bewoning
+import com.ritense.valtimo.portal.haalcentraal.brp.domain.persoon.Persoon
 
 interface HaalCentraalBrpService {
     suspend fun getPersoon(authentication: CommonGroundAuthentication): Persoon?
+    suspend fun getBewoningen(authentication: CommonGroundAuthentication): Bewoning ?
```
Comment (ivo-ritense) [inline haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/service/HaalCentraalBrpService.kt:24]: that space is confusing

File: haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/client/HaalCentraalBrpClient.kt [hunk 3]
```diff
@@ -37,4 +39,18 @@ class HaalCentraalBrpClient(
...
             .awaitBody()
     }
+
+    suspend fun getBewoningen(bsn: String): Bewoning {
+        return haalCentraalClientProvider.webClient()
+            .get()
+            .uri {
+                val uriBuilder = it.path("/bewoning/bewoningen")
+                    .queryParam("burgerservicenummer", bsn)
+                    .queryParam("peilDatum", LocalDate.now())
```
Comment (ivo-ritense) [inline haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/client/HaalCentraalBrpClient.kt:49]: Did you test/check if the date format this results in is supported by the API?

Comment (Rick-Ritense) [reply haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/client/HaalCentraalBrpClient.kt:49]: Yes I did. It matches with the postman calls

File: haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/service/HaalCentraalBrpService.kt (hunk 2 above)
Comment (Rick-Ritense) [reply haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/service/HaalCentraalBrpService.kt]: That is a weird one indeed. Fixed it

File: haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/client/HaalCentraalBrpClient.kt [hunk 4]
```diff
@@ -37,4 +39,18 @@ class HaalCentraalBrpClient(
             .retrieve()
             .awaitBody()
     }
+
+    suspend fun getBewoningen(bsn: String): Bewoning {
+        return haalCentraalClientProvider.webClient()
+            .get()
+            .uri {
+                val uriBuilder = it.path("/bewoning/bewoningen")
```
[the remark this answers is not in the document]
Comment (Rick-Ritense) [reply haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/client/HaalCentraalBrpClient.kt:47]: Nee de totale url is iets als 

https://www.haalcentraal.nl/haalcentraal/api/bewoning/bewoningen?peildatum=2022-04-11&burgerservicenummer=999991292

Daar komt dus geen brp in voor

[the remark this answers is not in the document]
Comment (Klaas-Ritense) [reply haal-centraal/src/main/kotlin/com/ritense/valtimo/portal/haalcentraal/brp/client/HaalCentraalBrpClient.kt:47]: Ach ja ik zie het. Prima

**Review by Klaas-Ritense — APPROVED** [review]

**Pull request merged**